# Solaris-specific patch. Not going upstream.
# Covers Solaris-specific configuration and makes it compatible with nroff.
--- rsyslog/tools/rsyslogd.8	2011-04-18 06:36:33.000000000 -0700
+++ rsyslog/tools/rsyslogd.8	2014-10-14 09:41:01.559928606 -0700
@@ -1,7 +1,8 @@
+'\" te
 .\" Copyright 2004-2008 Rainer Gerhards and Adiscon for the rsyslog modifications
 .\" May be distributed under the GNU General Public License
 .\"
-.TH RSYSLOGD 8 "29 July 2008" "Version 3.21.1" "Linux System Administration"
+.TH RSYSLOGD 1M "16 May 2012" "Version 6.2.0" "System Administration Commands"
 .SH NAME
 rsyslogd \- reliable and extended syslogd 
 .SH SYNOPSIS
@@ -44,17 +45,16 @@
 unix domain sockets enables this utility to support both local
 and remote logging.
 
-.B Note that this version of rsyslog ships with extensive documentation in html format.
-This is provided in the ./doc subdirectory and probably
-in a separate package if you installed rsyslog via a packaging system.
+.B "Note that this version of rsyslog ships with extensive documentation in html format."
+This is provided in the /usr/share/doc/rsyslog directory.
 To use rsyslog's advanced features, you
 .B need
 to look at the html documentation, because the man pages only cover
 basic aspects of operation.
-.B For details and configuration examples, see the rsyslog.conf (5)
-.B man page and the online documentation at http://www.rsyslog.com/doc
+.B "For details and configuration examples, see the rsyslog.conf (4)"
+.B "man page and the online documentation at http://www.rsyslog.com/doc"
 
-.BR Rsyslogd (8)
+.BR rsyslogd (1M)
 is derived from the sysklogd package which in turn is derived from the
 stock BSD sources.
 
@@ -93,9 +93,9 @@
 
 .LP
 .SH OPTIONS
-.B Note that in version 3 of rsyslog a number of command line options
-.B have been deprecated and replaced with config file directives. The
-.B -c option controls the backward compatibility mode in use.
+.B "Note that in version 3 of rsyslog a number of command line options"
+.B "have been deprecated and replaced with config file directives. The"
+.B "-c option controls the backward compatibility mode in use."
 .TP
 .BI "\-A"
 When sending UDP messages, there are potentially multiple paths to
@@ -132,8 +132,8 @@
 is the rsyslog version that it shall be
 compatible with. Using -c0 tells rsyslog to be command-line compatible
 to sysklogd, which is the default if -c is not given.
-.B Please note that rsyslogd issues warning messages if the -c3
-.B command line option is not given.
+.B "Please note that rsyslogd issues warning messages if the -c3"
+.B "command line option is not given."
 This is to alert you that your are running in compatibility
 mode. Compatibility mode interferes with your rsyslog.conf commands and
 may cause some undesired side-effects. It is meant to be used with a
@@ -170,7 +170,8 @@
 Avoid auto-backgrounding.  This is needed especially if the
 .B rsyslogd
 is started and controlled by
-.BR init (8).
+.BR init (1M),
+which is not the case on Solaris.
 .TP
 .B "\-N " "level"
 Do a coNfig check. Do NOT run in regular mode, just check configuration
@@ -181,9 +182,6 @@
 not specifying the -N option at all (so this makes limited sense) and
 1 actually activates the code. Later, higher levels will mean more
 verbosity (this is a forward-compatibility option).
-.B rsyslogd
-is started and controlled by
-.BR init (8).
 .TP
 .BI "\-q " "add hostname if DNS fails during ACL processing"
 During ACL processing, hostnames are resolved to IP addresses for
@@ -257,7 +255,7 @@
 operation and should only be done when actually necessary. Actually, it is
 a rsyslgod stop immediately followed by a restart. Future versions will remove
 this restart functionality of HUP (it will go away in v5). So it is advised to use
-HUP only for closing files, and a "real restart" (e.g. /etc/rc.d/rsyslogd restart)
+HUP only for closing files, and a "real restart" (e.g. svcadm restart svc:/system/system-log:rsyslog)
 to activate configuration changes.
 .TP
 .B TERM ", " INT ", " QUIT
@@ -330,12 +328,10 @@
 The file containing the process id of 
 .BR rsyslogd .
 .TP
-.I prefix/lib/rsyslog
+.I /usr/lib/rsyslog
 Default directory for
 .B rsyslogd
-modules. The
-.I prefix
-is specified during compilation (e.g. /usr/local).
+modules.
 .SH ENVIRONMENT
 .TP
 .B RSYSLOG_DEBUG
@@ -378,20 +374,42 @@
 .B RSYSLOG_MODDIR
 Provides the default directory in which loadable modules reside.
 .PD
-.SH BUGS
-Please review the file BUGS for up-to-date information on known
-bugs and annoyances.
 .SH Further Information
 Please visit
 .BR http://www.rsyslog.com/doc
 for additional information, tutorials and a support forum.
+.SH ATTRIBUTES
+.sp
+.LP
+See \fBattributes\fR(5) for descriptions of the following attributes:
+.sp
+
+.sp
+.TS
+tab() box;
+cw(2.75i) |cw(2.75i) 
+lw(2.75i) |lw(2.75i) 
+.
+ATTRIBUTE TYPEATTRIBUTE VALUE
+_
+Availabilitypkg:/system/rsyslog
+_
+Servicesvc:/system/system-log:rsyslog
+.TE
+
+.SH Solaris Usage
+The rsyslog instance of the system-log service is not enabled by default.
+To enable it, first "svcadm disable svc:/system/system-log:default", then
+"svcadm enable svc:/system/system-log:rsyslog".  If you wish to enable log
+rotation, then add each file referenced in
+.BR rsyslog.conf(4)
+to /etc/logadm.conf.
 .SH SEE ALSO
-.BR rsyslog.conf (5),
+.BR rsyslog.conf (4),
 .BR logger (1),
-.BR syslog (2),
-.BR syslog (3),
-.BR services (5),
-.BR savelog (8)
+.BR syslog (3C),
+.BR services (4)
+.BR logadm.conf (4)
 .LP
 .SH COLLABORATORS
 .B rsyslogd
